src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -91,7 +91,7 @@ public function prepend(ContainerBuilder $container): void
     public function load(array $configs, ContainerBuilder $container): void
     {
         if (!array_filter($configs)) {
-            $hint = class_exists(InstallRecipesCommand::class) ? 'Try running "composer symfony:recipes:install symfony/security-bundle".' : 'Please define your settings for the "security" config section.';
+            $hint = class_exists(InstalledVersions::class) && InstalledVersions::isInstalled('symfony/flex') ? 'Try running "composer symfony:recipes:install symfony/security-bundle".' : 'Please define your settings for the "security" config section.';
```

Comment:
can you try using `$container->getReflectionClass(InstallRecipesCommand::class, false)` instead? does it behave as expected then?

Reply:
@nicolas-grekas just tried it, `$container->getReflectionClass(InstallRecipesCommand::class, false)` returns null. The ClassExistenceResource::isFresh method appears to be throwing the same error about the BaseCommand class not being found.

Reply:
sorry, it returns or it throws? if it returns, that's what we want, no?

Reply:
@nicolas-grekas sorry, updated my comment to make it clearer. It returns null even when the class exists. We want it to return true so that we get the nicer message about running `composer symfony:recipes:install symfony/security-bundle`

Reply:
The behavior I'm expecting is:
- return a ReflectionClass when the class *and its parent* exist
- return null when the class OR its parent does not exist

but then we'd need another check to see if flex is installed, so let's go this way